### PR TITLE
Increase maxRetry from 1 to 2

### PR DESCRIPTION
**Pull request by blindpirate**
Before we implement flaky test quarantine.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test and merge

File: build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts [hunk 1]
```diff
@@ -242,7 +242,7 @@ fun configureTests() {
         if (BuildEnvironment.isCiServer) {
             configureRerun()
             retry {
-                maxRetries.convention(1)
+                maxRetries.convention(2)
```
Comment (wolfs) [inline build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts:245]: Is this configuration for performance tests as well? Then we probably should reduce it to 1 there. Same probably for smoke tests. WDYT?

Comment (blindpirate) [reply build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts]: `PerformanceTest` is not visible in this scope, so I did the judgement by project name.

**Review by wolfs — APPROVED** [review]

**Comment by blindpirate** [comment]
@bot-gradle cancel

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**